core/server/api/settings.js

```diff
@@ -103,6 +103,15 @@ canEditAllSettings = function (settingsInfo, options) {
                 ));
             }
 
+            if (setting.key === 'activeTheme') {
```

Comment:
Hmm how about 
`if (setting.type === 'core' && !(options.context && options.context.internal)) {` replaced by
 `if ((setting.type === 'core' || setting.type === 'theme') && !(options.context && options.context.internal)) {`?

Or similar e.g. we make `activeTheme` part of `core`.
The goal is to have less extra checks here.